### ISSUE TITLE
iotedge 1.2.7: EdgeHub loaded certificate with expiration date of a month (#2)

### ISSUE THREAD

**Issue by alaendle** [Azure/iotedge#6158]
There still seems to be an issue with certificate renewal - at least for us the new server certificate was only loaded after a manual restart.
`sudo docker logs edgeHub | grep certificate` shows that:
```
<6> 2022-01-24 07:26:48.892 +00:00 [INF] - Installing certificates [CN=bh0vm00001CA.ca:02/06/2036 16:44:14],[CN=Azure_IoT_Hub_Intermediate_Cert:02/06/2036 16:42:24],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23] to Root
<6> 2022-01-24 07:26:48.924 +00:00 [INF] - Installing certificates [CN=bh0vm00001CA.ca:02/06/2036 16:44:14],[CN=Azure_IoT_Hub_Intermediate_Cert:02/06/2036 16:42:24],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23] to Root
<6> 2022-01-24 07:26:50.624 +00:00 [INF] - Loaded server certificate with expiration date of "2022-02-23T07:26:48.0000000+00:00"
<6> 2022-01-24 07:26:51.464 +00:00 [INF] - Scheduling server certificate renewal for "2022-02-23T07:24:18.0000986Z".
<6> 2022-02-23 07:24:18.007 +00:00 [INF] - Restarting process to perform server certificate renewal.
```
from the logs it seems like edgeHub really restarts, but the modules nonetheless could only reconnect after the complete iotedge service was restarted. Generating the following logs:
```
<6> 2022-02-24 11:18:21.704 +00:00 [INF] - Installing certificates [CN=bh0vm00001CA.ca:02/06/2036 16:44:14],[CN=Azure_IoT_Hub_Intermediate_Cert:02/06/2036 16:42:24],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23] to Root
<6> 2022-02-24 11:18:21.742 +00:00 [INF] - Installing certificates [CN=bh0vm00001CA.ca:02/06/2036 16:44:14],[CN=Azure_IoT_Hub_Intermediate_Cert:02/06/2036 16:42:24],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23],[CN=Azure_IoT_Hub_CA_Cert:02/06/2036 16:42:23] to Root
<6> 2022-02-24 11:18:25.870 +00:00 [INF] - Loaded server certificate with expiration date of "2022-03-26T11:18:21.0000000+00:00"
<6> 2022-02-24 11:18:27.314 +00:00 [INF] - Scheduling server certificate renewal for "2022-03-26T11:15:51.0000882Z".
```

Since a lot of similar issues are open I hope the problem is already known and under investigation. But if you think this should not also occur in conjunction with version 1.2.7 or if you need further information in general, please don't hesitate to contact me. For now I kept short - and thanks in advance for your support.

## Expected Behavior
Modules should be able to reconnect.

## Current Behavior
No module was able to reconnect without a restart of the complete iot edge service (`iotedge restart`).

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. Just wait one month?

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```
Configuration checks (aziot-identity-service)
---------------------------------------------
√ keyd configuration is well-formed - OK
√ certd configuration is well-formed - OK
√ tpmd configuration is well-formed - OK
√ identityd configuration is well-formed - OK
√ daemon configurations up-to-date with config.toml - OK
√ identityd config toml file specifies a valid hostname - OK
√ aziot-identity-service package is up-to-date - OK
‼ host time is close to reference time - Warning
    Could not query NTP server
√ production readiness: identity certificates expiry - OK
√ preloaded certificates are valid - OK
√ keyd is running - OK
√ certd is running - OK
√ identityd is running - OK
√ read all preloaded certificates from the Certificates Service - OK
√ read all preloaded key pairs from the Keys Service - OK
√ ensure all preloaded certificates match preloaded private keys with the same ID - OK

Connectivity checks (aziot-identity-service)
--------------------------------------------
× host can connect to and perform TLS handshake with iothub AMQP port - Error
    Could not connect to ioth-bhp-edge-dev.azure-devices.net : could not complete TLS handshake
√ host can connect to and perform TLS handshake with iothub HTTPS / WebSockets port - OK
× host can connect to and perform TLS handshake with iothub MQTT port - Error
    Could not connect to ioth-bhp-edge-dev.azure-devices.net : could not complete TLS handshake

Configuration checks
--------------------
√ aziot-edged configuration is well-formed - OK
√ configuration up-to-date with config.toml - OK
√ container engine is installed and functional - OK
√ configuration has correct URIs for daemon mgmt endpoint - OK
‼ aziot-edge package is up-to-date - Warning
    Error while fetching latest versions of edge components: could not send HTTP request
√ container time is close to host time - OK
√ DNS server - OK
√ production readiness: logs policy - OK
√ production readiness: Edge Agent's storage directory is persisted on the host filesystem - OK
√ production readiness: Edge Hub's storage directory is persisted on the host filesystem - OK
√ Agent image is valid and can be pulled from upstream - OK
‼ proxy settings are consistent in aziot-edged, aziot-identityd, moby daemon and config.toml - Warning
    The proxy setting for IoT Edge Agent "http://rb-proxy-sl.bosch.com:8080", IoT Edge Daemon "", IoT Identity Daemon "", and Moby "" may need to be identical.
```
</details>

### Device Information
* Host OS [e.g. Ubuntu 18.04, Windows Server IoT 2019]: RHEL 7.9
* Architecture [e.g. amd64, arm32, arm64]: amd64
* Container OS [e.g. Linux containers, Windows containers]: Linux

### Runtime Versions
* aziot-edged [run `iotedge version`]: iotedge 1.2.7
* Edge Agent [image tag (e.g. 1.0.0)]: 1.2.7
* Edge Hub [image tag (e.g. 1.0.0)]: 1.2.7
* Docker/Moby [run `docker version`]: 20.10.11+azure-3

**Comment by lfitchett** [comment]
@alaendle The edgehub server certificate expiry is set to one month. The eh server certificate will automatically be rotated when it is close to expiry. This will trigger an edgehub restart automatically. 

You mention 
> "from the logs it __seems like__ edgeHub really restarts, but the modules nonetheless could only reconnect after the complete iotedge service was restarted. Generating the following logs:"  

Is edgehub fully operational after the manual restart, as in data is flowing to the cloud? Can you attach the edgehub logs from before the manual restart?

**Comment by alaendle** [comment]
Thanks @lfitchett for analyzing this issue - I've added the log (beginning at the point of the `process to perform server certificate renewal` and just showing the following minutes).

[edgeHub_certificate_renewal.txt](https://github.com/Azure/iotedge/files/8138796/edgeHub_certificate_renewal.txt)

Please let me know if I could contribute further information.

**Comment by lfitchett** [comment]
@alaendle the logs you sent look good to me, I don't see any reason why edgehub wouldn't have restarted. Were there any errors listed in edged or edgeAgent? 

Also, in the original snippet you sent, the device was offline for a month? The date of the logs goes from 1/24 to 2/23

**Comment by lfitchett** [comment]
@alaendle This is the same as issue #6008. It has been fixed in our newest release from yesterday, 1.2.8. 
https://github.com/Azure/azure-iotedge/releases

Please re-open this issue if the problem repeats on 1.2.8.
